interface_service/src/bucket.py

```diff
@@ -0,0 +1,43 @@
+"""
+Provides download and upload functionality
+For interfacing with MinIO buckets
+"""
+from minio import Minio
+from config import settings
+
+
+def authenticate():
+    """Authenticates to MinIO.
+
+    Returns:
+        Minio: The MinIO client object.
+    """
+
+    # Get credentials from env variables
+    return Minio(
+        settings["MINIO_ENDPOINT"],
+        access_key=settings["MINIO_ACCESS_KEY"],
+        secret_key=settings["MINIO_SECRET_KEY"],
+        # Without this, certificates are required
+        secure=False
+    )
+
+
+def upload_model():
+    """Uploads a file to the MinIO bucket.
+    """
+    client = authenticate()
```

Comment:
Authentication should be done once during program initialization, not on every request to the object storage's API. The authenticated client should be either stored as a singleton or passed as an argument to the functions that upload/download the model.